docs/basics/101-140-filesystem.rst

```diff
@@ -0,0 +1,764 @@
+
+.. runrecord:: _examples/DL-101-140-160
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   # Install a subdataset - the content is irrelevant, so why not a cloud :)
+   $ datalad install -d . \
+    --source https://github.com/datalad-datasets/disneyanimation-cloud.git \
+    cloud
+
+To uninstall the dataset, use
+
+.. runrecord:: _examples/DL-101-140-161
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad uninstall cloud
+
+Note that the dataset is still known in the dataset, and not completely removed.
+
+In case one wants to fully delete a subdataset from a dataset, the
+:command:`datalad remove` command (:manpage:`datalad-remove` manual) is
+relevant [#f1]_.
+It needs a pointer to the root of the superdataset with the ``-d/--dataset``
+flag, a path to the subdataset to be removed, and optionally a commit message
+(``-m/--message``) or recursive specification (``-r/--recursive``).
+To remove a subdataset, we will install the uninstalled subdataset again, and
+subsequently remove it with the :command:`datalad remove` command:
+
+.. runrecord:: _examples/DL-101-140-163
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad install cloud
+   # delete the subdataset
+   $ datalad remove -m "remove obsolete subds" -d . cloud
+
+Finally, let's clean up:
+
+.. runrecord:: _examples/DL-101-140-164
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ git reset --hard HEAD~2
+
+Deleting a superdataset
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If for whatever reason you at one point tried to remove a DataLad dataset,
+whether with a GUI or the command line call ``rm -rf <directory>``, you likely
+have seen permission denied errors such as
+
+.. code-block::
+
+    rm: cannot remove '<directory>/.git/annex/objects/Mz/M1/MD5E-s422982--2977b5c6ea32de1f98689bc42613aac7.jpg/MD5E-s422982--2977b5c6ea32de1f98689bc42613aac7.jpg': Permission denied
+    rm: cannot remove '<directory>/.git/annex/objects/FP/wv/MD5E-s543180--6209797211280fc0a95196b0f781311e.jpg/MD5E-s543180--6209797211280fc0a95196b0f781311e.jpg': Permission denied
+    [...]
+
+This error indicates that there is write protected content within ``.git`` that
+cannot not be deleted. What is this write-protected content? It's the file content
+stored in the object tree of Git-annex. If you want, you can re-read section on
+:ref:`symlink` to find out how Git-annex revokes write permission for the user
+to protect the file content given to it. To remove a dataset with annexed content
+one has to regain write permissions to everything in the dataset. This is done
+with the `chmod <https://en.wikipedia.org/wiki/Chmod>`_ command::
+
+    chmod -R u+w <dataset>
+
+This *recursively* (``-R``, i.e. throughout all files and (sub)directories) gives users
+(``u``) write permissions (``+w``) for the dataset.
+
+Afterwards, ``rm -rf <dataset>`` will succeed. Be aware though that this will
+irretrievable delete the dataset, it's contents, and it's history.
```

Comment:
`datalad remove <superdataset>` (likely with `--nocheck` to be faster, and possibly with `--recursive` in case of subdatasets) is the dedicated DataLad command to achieve this. However, one (or at least me) often performs an `rm -rf` first, sees the error shown above, regrets doing so, and has to fix up exactly in this way -- hence having this info here is crucial.

Just as `rm -rf .` will not work, `datalad remove .` is not possible -- removal only works from a working directory outside the dataset.